diff --git a/node_modules/react-native-mmkv/lib/index.d.ts b/node_modules/react-native-mmkv/lib/index.d.ts
index 637e3d0..498e21d 100644
--- a/node_modules/react-native-mmkv/lib/index.d.ts
+++ b/node_modules/react-native-mmkv/lib/index.d.ts
@@ -1,6 +1,14 @@
-export type { MMKV } from './specs/MMKV.nitro';
-export type { Configuration, Mode } from './specs/MMKVFactory.nitro';
-export { createMMKV } from './createMMKV/createMMKV';
+import { MMKV as MMKVInterface } from './specs/MMKV.nitro';
+import { Configuration, Mode } from './specs/MMKVFactory.nitro';
+import { createMMKV } from './createMMKV/createMMKV';
+
+export interface MMKV extends MMKVInterface {}
+export class MMKV {
+  constructor(configuration?: Configuration);
+}
+
+export type { Configuration, Mode };
+export { createMMKV };
 export { existsMMKV } from './existsMMKV/existsMMKV';
 export { deleteMMKV } from './deleteMMKV/deleteMMKV';
 export { useMMKV } from './hooks/useMMKV';
diff --git a/node_modules/react-native-mmkv/lib/index.js b/node_modules/react-native-mmkv/lib/index.js
index 315c604..bbad4d7 100644
--- a/node_modules/react-native-mmkv/lib/index.js
+++ b/node_modules/react-native-mmkv/lib/index.js
@@ -1,5 +1,18 @@
-// The create function
-export { createMMKV } from './createMMKV/createMMKV';
+import { createMMKV } from './createMMKV/createMMKV';
+
+export class MMKV {
+  constructor(configuration) {
+    const instance = createMMKV(configuration);
+    if (instance != null && typeof instance.delete !== 'function') {
+      instance.delete = function (key) {
+        return this.remove(key);
+      };
+    }
+    return instance;
+  }
+}
+
+export { createMMKV };
 // Exists + Delete
 export { existsMMKV } from './existsMMKV/existsMMKV';
 export { deleteMMKV } from './deleteMMKV/deleteMMKV';
diff --git a/node_modules/react-native-mmkv/src/index.ts b/node_modules/react-native-mmkv/src/index.ts
index f059d8a..62a5a4e 100644
--- a/node_modules/react-native-mmkv/src/index.ts
+++ b/node_modules/react-native-mmkv/src/index.ts
@@ -1,9 +1,22 @@
-// All types
-export type { MMKV } from './specs/MMKV.nitro'
-export type { Configuration, Mode } from './specs/MMKVFactory.nitro'
+import { createMMKV } from './createMMKV/createMMKV'
+import type { Configuration, Mode } from './specs/MMKVFactory.nitro'
+import type { MMKV as MMKVInterface } from './specs/MMKV.nitro'
 
-// The create function
-export { createMMKV } from './createMMKV/createMMKV'
+export interface MMKV extends MMKVInterface {}
+export class MMKV {
+  constructor(configuration?: Configuration) {
+    const instance = createMMKV(configuration)
+    if (instance != null && typeof (instance as any).delete !== 'function') {
+      ;(instance as any).delete = function (key: string) {
+        return this.remove(key)
+      }
+    }
+    return instance as any
+  }
+}
+
+export type { Configuration, Mode }
+export { createMMKV }
 
 // Exists + Delete
 export { existsMMKV } from './existsMMKV/existsMMKV'
